### PR TITLE
Magnetism

### LINKED ISSUE #133: Enchantment that collects drops
### Enchantment name

Magnetism

### Enchantment Info

Magnetism would instantly collect ores and other items when you mine.
Would be able to be applied to pickaxes.

Would only have 1 level as effect would not really differ

### Obtainability

- [X] Enchanted Books
- [ ] Enchantment Table (Randomly applied)
- [X] Vendor Trades
- [ ] Dungeon Loot
- [ ] Other (Details in Additional Info)

### Additional info

_No response_

### Agreements

- [X] I have searched for and ensured there isn't already an open suggestion similar to this.
- [X] I have ensured the mob I'm suggesting isn't already on the server.

### PR DESCRIPTION
**Pull request by desmont323**
add code for the enchant Magnetism

Closes #133

### CODE REVIEW

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/CustomEnchants.kt [hunk 1]
```diff
@@ -19,14 +19,20 @@ object CustomEnchants {
     val BIRD_SWATTER = EnchantmentWrapper("birdswatter", "Bird Swatter", 5, EnchantmentTarget.WEAPON, color(0,220,60))
     val JAW_BREAKER = EnchantmentWrapper("jawbreaker", "Jaw Breaker", 3, EnchantmentTarget.WEAPON, color(150,20,150))
     val BANE_OF_KUONGATARI = EnchantmentWrapper("baneofkuongatari", "Bane of Kuongatari", 1, EnchantmentTarget.WEAPON, color(0,200,80))
-
+    val Magnetism = EnchantmentWrapper("magnetism", "Magnetism", 1, EnchantmentTarget.TOOL, color(200,200,80))
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/CustomEnchants.kt:22]: Keep all caps like the others, we use that to show immutable top-level values.

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/anticheese/AntiCheeseListener.kt [hunk 2]
```diff
@@ -49,7 +49,6 @@ class AntiCheeseListener : Listener {
             }
         }
     }
-
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/anticheese/AntiCheeseListener.kt:52]: Dont remove this space :sob:

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        if (item.containsEnchantment(Magnetism)){
+
+            val loc = block.location
+
+            mineInAbyss.schedule {
+                waitFor(1)
+
+                val entities = player.world.getNearbyEntities(loc, 4.0, 4.0, 4.0)
+
+                for (entitie in entities) {
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt:27]: `entity` is the singular form

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+                    if (entitie is Item) {
+                        val item: Item = entitie
+                        val fullInv = player.inventory.firstEmpty() == -1
+                        if (!fullInv){
+                            player.inventory.addItem(item.itemStack)
```
Comment (0ffz) [inline mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt:32]: I believe addItem retuns any leftover items, you can update the item on ground to have the correctly adjusted stack size depending on this

**Review by 0ffz — CHANGES_REQUESTED** [review]

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt (hunk 3 above)
Comment (desmont323) [reply mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt:27]: sorry for that im French and have dysorthographia ;(

File: mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt (hunk 4 above)
Comment (desmont323) [reply mineinabyss-features/src/main/kotlin/com/mineinabyss/enchants/enchantments/MagnetismListener.kt]: i changed the code for this a bit but im not sure about how i could directly change the stack size on the ground

**Pull request closed without merging**